.github/workflows/ubuntu_18_04.yml

```diff
@@ -0,0 +1,73 @@
+name: Kinetic & Melodic ROS CI
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os_distro: [[ubuntu-16.04, kinetic], [ubuntu-18.04, melodic]]
+    runs-on: ${{ matrix.os_distro[0] }}
+    env:
+      ROS_CI_DESKTOP: "`lsb_release -cs`"  # e.g. [trusty|xenial|...]
+      ROS_DISTRO: ${{ matrix.os_distro[1] }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: catkin_ws/src/ros1_lifecycle
+      ### ROS ###
+      - name: Setup for ROS ${{ matrix.os_distro[1] }} install
+        run: |
+            sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
+            sudo apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
+            sudo apt-get update -qq
+            sudo apt-get install dpkg
+      - name: Install ROS basic packages
+        run: |
+            sudo apt-get install -y python-catkin-pkg
+            sudo apt-get install -y python-catkin-tools
+            sudo apt-get install -y python-rosdep
+            sudo apt-get install -y python-rospkg
+            sudo apt-get install -y python-wstool
+            sudo apt-get install -y ros-$ROS_DISTRO-ros-base
+            source /opt/ros/$ROS_DISTRO/setup.bash
+            # Prepare rosdep to install dependencies.
+            sudo rosdep init
+            rosdep update
+      - name: Install ROS additional packages with rosdep
+        run: |
+            source /opt/ros/$ROS_DISTRO/setup.bash
+            cd catkin_ws
+            rosdep install --from-paths src --ignore-src -r -s  # do a dry-run first
+            rosdep install --from-paths src --ignore-src -r -y
+      - name: Catkin build install
+        run: |
+          source /opt/ros/$ROS_DISTRO/setup.bash
+          mkdir -p catkin_ws/src
+          cd catkin_ws
+          catkin config --install --cmake-args -DCMAKE_BUILD_TYPE=Release
```

Comment:
Release build will remove asserts, so I would go for a debug build here.

Reply:
How about the devel build is the Debug build?  That's the build that ends up running the test that contains asserts.